bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -1,4 +1,8 @@
 /**
```

Comment:
I've optimize synchronizations. Also I have remove status map from rule engine and put it into runtime rule. I don't see why we have to hold a rules' status into separate map and do unduly synchronization between runtime rules and status map. Each  runtime rule must have a status and I think it must be a field of it.